optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/entity/pillar/PillarSelectorConfig.java

```diff
@@ -52,10 +57,18 @@ public void setEntitySelectorConfig(EntitySelectorConfig entitySelectorConfig) {
         this.entitySelectorConfig = entitySelectorConfig;
     }
 
+    /**
+     * @see PillarType and its uses in pillar move selectors.
+     */
+    @Deprecated(/* forRemoval = true */)
```

Comment:
Please add a line in the description of this jira so we don't forget to remove this in 8.0:
  https://issues.jboss.org/browse/PLANNER-750

Please also create a PR to add a note in the upgrade recipe:
  https://www.optaplanner.org/download/upgradeRecipe/upgradeRecipe7.html
  https://github.com/kiegroup/optaplanner-website/blob/master/download/upgradeRecipe/upgradeRecipe7.adoc